rpc/send.go

```diff
@@ -151,7 +151,19 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 				log.Infof("%s: sending request to %s: %+v", method, addr, args)
 			}
 			trace.Event(fmt.Sprintf("sending to %s", addr))
-			go sendOneFn(client, opts.Timeout, method, args, getReply(), done)
+
+			reply := getReply()
+
+			// Don't bother firing off a goroutine in the common case where a client
+			// is already healthy.
+			select {
+			case <-client.Healthy():
+				client.Go(method, args, reply, done)
```

Comment:
Good catch. Didn't notice that. The tests reliably pass. Unfortunately, function pointers can only be compared to `nil`. Let me see what else I can do.